### PR TITLE
sbcl: 2.2.4 -> 2.2.6

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes

Also add support for `sb-core-compression`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 179949` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by 7c6f434c** [comment]
@ofborg build lispPackages

**Comment by marsam** [comment]
I think aarch64-linux is failing because sb-simd. I'll take a look in the afternoon.

**Comment by marsam** [comment]
Fixed the build on aarch64-linux.
@GrahamcOfBorg build sbcl
@GrahamcOfBorg build lispPackages

File: pkgs/development/compilers/sbcl/common.nix [hunk 1]
```diff
@@ -21,7 +22,8 @@ stdenv.mkDerivation rec {
     inherit sha256;
   };
 
-  buildInputs = [texinfo];
+  nativeBuildInputs = [ texinfo ];
+  buildInputs = lib.optionals coreCompression [ zstd ];
```
Comment (sternenseemann) [inline pkgs/development/compilers/sbcl/common.nix:26]: In principle we could also support it for < 2.2.6 using `zlib` (although maybe not turn on by default).

**Comment by sternenseemann** [comment]
All of `quicklispPackages` builds with SBCL 2.2.6, at least without core compression enabled from my testing in #180059

**Comment by rapenne-s** [comment]
I started nixpkgs-review on it which seems very long. However it quickly triggers two errors on linux-x86_64

```
error: builder for '/nix/store/9vdslflp9mbi6wxncvc9i08d7b5rn3x4-clpm-0.4.1.drv' failed with exit code 1;
       last 10 log lines:
       > 71: (SB-IMPL::TOPLEVEL-INIT)
       > 72: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
       > 73: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
       > 74: (SB-IMPL::%START-LISP)
       >
       > unhandled condition in --disable-debugger mode, quitting
       > ;
       > ; compilation unit aborted
       > ;   caught 2 fatal ERROR conditions
       > ;   caught 18 WARNING conditions
       For full logs, run 'nix log /nix/store/9vdslflp9mbi6wxncvc9i08d7b5rn3x4-clpm-0.4.1.drv'.
error: builder for '/nix/store/qlax0myn7djpm2v3zbcsfaar7airkz1n-sbcl-2.0.8.drv' failed with exit code 1;
       last 10 log lines:
       > * make[1]: Leaving directory '/build/sbcl-2.0.8/contrib/sb-cover'
       > make: Leaving directory '/build/sbcl-2.0.8/contrib'
       >
       > WARNING! Some of the contrib modules did not build successfully or pass
       > their self-tests. Failed contribs:"
       >   sb-concurrency
       >
       > real      0m26.331s
       > user 0m22.647s
       > sys  0m3.565s
       For full logs, run 'nix log /nix/store/qlax0myn7djpm2v3zbcsfaar7airkz1n-sbcl-2.0.8.drv'.
```

Are these failures already expected or introduced by this change?

**Comment by sternenseemann** [comment]
@rapenne-s Can't reproduce this, I think this is just a case of the concurrency test suite being flaky on busy systems.

**Review by sternenseemann — APPROVED** [review]

**Pull request merged**